client/lib/media/utils.js

```diff
@@ -122,8 +123,13 @@ export function getFileExtension( media ) {
 	} else if ( media.extension ) {
 		extension = media.extension;
 	} else {
-		const pathname = urlLib.parse( media.URL || media.file || media.guid || '' ).pathname || '';
-		extension = path.extname( pathname ).slice( 1 );
+		try {
+			const pathname =
+				new URL( media.URL || media.file || media.guid || '', BASE_PARSE_URL ).pathname || '';
+			extension = path.extname( pathname ).slice( 1 );
+		} catch {
+			return '';
+		}
```

Comment:
same here…

```js
} else {
	const parsedPath = parseURL( media.URL || media.file || media.guid || '', BASE_PARSE_URL );
	const pathname = parsedPath.?pathname || '';

	extension = path.extname( pathname ).slice( 1 );
}
```

not sure exactly what behavior we're trying to handle in the `catch` statement

Reply:
This else block handles the situation where we weren't passed a string, but rather an object with one of several well-known properties. It tries to compose a URL from that and look for the extension, and returns an empty string otherwise.

It will return an empty string if it can't find any of the properties, or if there's no extension. It will throw, be caught, and return an empty string if the URL is invalid.